drivers/wdm/audio/backpln/portcls/pin_wavepci.cpp

```diff
@@ -542,6 +545,45 @@ CPortPinWavePci::HandleKsProperty(
     return Status;
 }
 
+NTSTATUS
+NTAPI
+CPortPinWavePci::HandleKsReset(
+    IN PIRP Irp)
+{
+    KSRESET ResetValue;
+    NTSTATUS Status;
+
+    Status = KsAcquireResetValue(Irp, &ResetValue);
+    DPRINT("Status %x Value %u\n", Status, ResetValue);
+    /* check for success */
+    if (NT_SUCCESS(Status))
+    {
+        //determine state of reset request
+        if (ResetValue == KSRESET_BEGIN)
+        {
+            // start reset process
+            // incoming read/write requests will be rejected
+            m_ResetState = KSRESET_BEGIN;
+
+            // cancel existing buffers
+            m_IrpQueue->CancelBuffers();
+        }
+        else if (ResetValue == KSRESET_END)
+        {
+            // end of reset process
+            m_ResetState = KSRESET_END;
+        }
+    }
+
+    if (Status != STATUS_PENDING)
```

Comment:
STATUS_PENDING can't happen here so I'd recommend against this check, to avoid giving the impression that it can. Perhaps assert `Status != STATUS_PENDING`.